units/XNL0403/XNL0403_Script.lua

```diff
@@ -92,7 +98,9 @@ XNL0403 = Class(NExperimentalHoverLandUnit, SlowHover) {
         end
     end,
 
-    PlayMovementSmokeEffects = function(self, type)
+    ---@param self XNL0403
+    ---@param damageType DamageType unused
+    PlayMovementSmokeEffects = function(self, damageType)
```

Comment:
You did not refactor the rest of the code in the function when changing the parameter name. It is also not unused and isn't a `DamageType`.
```suggestion
    ---@param type HorizontalMovementState
    PlayMovementSmokeEffects = function(self, type)
```